homeassistant/components/twinkly/light.py

```diff
@@ -0,0 +1,224 @@
+        # if the device is currently offline.
+        # They are expected to be updated using the device_info.
+        self.__name = name
+        self.__model = model
+
+        # Set default state before any update
+        self._is_on = False
+        self._brightness = 0
+        self._is_available = False
+        self._attributes = {ATTR_HOST: client.host}
+
+    @property
+    def supported_features(self):
+        """Get the features supported by this entity."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def should_poll(self) -> bool:
+        """Get a boolean which indicates if this entity should be polled."""
+        return True
+
+    @property
+    def available(self) -> bool:
+        """Get a boolean which indicates if this entity is currently available."""
+        return self._is_available
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Id of the device."""
+        return self._id
+
+    @property
+    def name(self) -> str:
+        """Name of the device."""
+        return self.__name if self.__name else "Twinkly light"
+
+    @property
+    def model(self) -> str:
+        """Name of the device."""
+        return self.__model
+
+    @property
+    def icon(self) -> str:
+        """Icon of the device."""
+        return "mdi:string-lights"
+
+    @property
+    def device_info(self) -> Optional[Dict[str, Any]]:
+        """Get device specific attributes."""
+        return (
+            {
+                "identifiers": {(DOMAIN, self._id)},
+                "name": self.name,
+                "manufacturer": "LEDWORKS",
+                "model": self.model,
+            }
+            if self._id
+            else None  # device_info is available only for entities configured from the UI
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self._is_on
+
+    @property
+    def brightness(self) -> Optional[int]:
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def state_attributes(self) -> dict:
+        """Return device specific state attributes."""
+
+        attributes = self._attributes
+
+        # Make sure to update any normalized property
+        attributes[ATTR_HOST] = self._client.host
+        attributes[ATTR_BRIGHTNESS] = self._brightness
+
+        return attributes
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = int(int(kwargs[ATTR_BRIGHTNESS]) / 2.55)
+
+            # If brightness is 0, the twinkly will only "disable" the brightness,
+            # which means that it will be 100%.
+            if brightness == 0:
+                await self._client.set_is_on(False)
+                return
+
+            await self._client.set_brightness(brightness)
+
+        await self._client.set_is_on(True)
```

Comment:
No need to convert this to a `DataUpdateCoordinator` right now as mentioned previously, but you may want to request an update here and in `async_turn_off` so the state change is available immediately instead of waiting for the next `schedule_interval` to pass.

Reply:
Actually my initial plan was to do that, but I noticed that after each command (on / off), home assistant automatically requests an update of the device, so it would make a duplicated call.

Reply:
I saw that there is a `schedule_update_ha_state` in the base class `Entity` that would fit, but apparently there is no debouncing/deduplication in place ... not sure if you had something else in mind?